forest/daemon/src/daemon.rs

```diff
@@ -398,18 +391,51 @@ pub(super) async fn start(config: Config, detached: bool) {
     info!("Forest finish shutdown");
 }
 
+async fn prompt_and_fetch_snapshot(config: &mut Config) {
+    if !config.client.download_snapshot {
+        if let false = Confirm::with_theme(&ColorfulTheme::default())
+            .with_prompt(
+                "Forest needs a snapshot to sync with the network. Would you like to download one now?",
+            )
+            .default(false)
+            .interact()
+            .expect("infallible. yes or no with default")
```

Comment:
This is not infallible. IO errors can happen and they're unrelated to the `default` value.

Reply:
Fixed and change to a match for error handling over expect